Socket.php

```diff
@@ -100,9 +280,44 @@ public static function transportFactory($socketUri)
                     ? 994
                     : 6667);
 
+        $parsed = array_merge([
+            'path'     => '',
+            'fragment' => '',
+            'query'    => '',
+            'user'     => null,
+            'pass'     => null
+        ], $parsed);
+        $complement =
+            ltrim($parsed['path'], '/') .
+            $parsed['fragment'] .
+            (empty($parsed['query'])?'':'?' . $parsed['query']);
+
+        $entity   = null;
+        $flags    = null;
+        $options  = [];
+
+        $pattern  = '/^((#|%23)?([^,?]+))(,([^?]+))?(\?(.*))?$/';
```

Comment:
You can name pattern in the regular expression, thus you will be able to refer to them with explicit name instead of abstract index. For instance: `/^(?<foo>\w+/` and then `$matches['foo']` instead of `$matches[1]`.